libmcount/mcount.c

```diff
@@ -1716,6 +1724,145 @@ static void mcount_script_init(enum uftrace_pattern_type patt_type)
 	strv_free(&info.cmds);
 }
 
+/* unlink socket file if it exists */
+static void unlink_socket_maybe(struct sockaddr_un *addr)
+{
+	if (unlink(addr->sun_path) == -1) {
+		if (errno != ENOENT) {
+			pr_dbg("cannot unlink %s\n", addr->sun_path);
+		}
+	}
+}
+
+static int agent_init(struct sockaddr_un *addr)
+{
+	int sfd;
+
+	if (mkdir(MCOUNT_AGENT_SOCKET_DIR, 0775) == -1) {
+		if (errno != EEXIST) {
+			pr_dbg("error creating run directory %s\n", MCOUNT_AGENT_SOCKET_DIR);
```

Comment:
Isn't it a critical error?  I wonder if it's a part of `socket_create`.

Reply:
It is a critical error, so I'm making it return.

However `socket_create()` doesn't bind to the socket, so it doesn't interact with the filesystem.
`socket_listen` does (I gather `bind()` and `listen()` calls there), but I feel like verifying that the directory exists prior to attempting socket creation/binding.

Reply:
I'm ok with checking EEXIST.  Other error code means it has a more serious error so it should fail.